pkg/notifier/decorators.go

```diff
@@ -41,37 +41,37 @@ func (w *withTimeout) Validate(ctx context.Context) error {
 	return nil
 }
 
-// WithRetry decorates the given Transport with retry logic using exponential backoff
-func WithRetry(transport Transport, maxRetries uint64, opts ...backoff.ExponentialBackOffOpts) Transport {
+type BackOff = backoff.BackOff
+
+// WithRetry decorates the given Transport with retry logic using the provided backoff
+func WithRetry(transport Transport, maxTries uint, backoffProvider func() BackOff) Transport {
```

Comment:
nit: for some reason the naming of `provider` seems like its overstating its role, but the more I think about alternatives the more its growing on me. maybe because we don't have a `type BackoffProvider`

Reply:
That works for me! I've renamed this to `factory` instead of `provider` and added a named type via 980491a